packages/tx-construction/src/types.ts

```diff
@@ -46,6 +47,8 @@ export interface InitializeTxProps {
   witness?: InitializeTxWitness;
   signingOptions?: Pick<SignTransactionOptions, 'additionalKeyPaths'>;
   handleResolutions?: HandleResolution[];
+  /** callback function that allows updating the transaction before input selection */
+  customizeCb?: (txBody: Readonly<TxBodyPreInputSelection>) => TxBodyPreInputSelection;
```

Comment:
It is likely that this function would need additional context, such as data resolved by tx builder providers (utxo, protocol parameters etc.). This would make it a little bit easier to extend in the future.
```suggestion
  type CustomizeCbProps = { txBody: Readonly<TxBodyPreInputSelection> };
  ...
  customizeCb?: (props: CustomizeCbProps) => TxBodyPreInputSelection;
```

Reply:
Done here: [083a766](https://github.com/input-output-hk/cardano-js-sdk/pull/1099/commits/083a766f966d4705b927dceef7309a970ffffa17)